src/libcore/task/task.rs

```diff
@@ -30,7 +30,7 @@ unsafe impl Send for TaskObj {}
 impl TaskObj {
     /// Create a `TaskObj` from a custom trait object representation.
     #[inline]
-    pub fn new<T: UnsafeTask>(t: T) -> TaskObj {
+    pub fn new<T: UnsafeTask + Send>(t: T) -> TaskObj {
```

Comment:
Nit: rather than repeating all the logic between `TaskObj` and `LocalTaskObj`, I'd make `TaskObj` into a single-field wrapper around `LocalTaskObj` that just adds `Send`-ability at the cost of requiring `Send` to construct (in `new`).

Reply:
The current way was inspired by the way it is done for `LocalWaker` & `Waker`. You're right. A single field struct is nicer

Reply:
`Waker` and `LocalWaker` are a bit different because they actually call different functions, but it's possible they could be combined further as well.

Reply:
Ok good point. Anyhow I think the current waker code is good. At closer inspection it really shows that a lot of considerations went into it.